tests/integration_tests/functional/test_signals.py

```diff
@@ -58,6 +66,50 @@ def test_sigbus_sigsegv(test_microvm_with_api, signum):
     assert metric_line["signals"][signum_str[signum]] == 1
 
 
+def test_sigxfsz_handler(test_microvm_with_api):
+    """Test intercepting and handling SIGXFSZ."""
+    microvm = test_microvm_with_api
+    microvm.spawn()
+
+    # We don't need to monitor the memory for this test.
+    microvm.memory_monitor = None
+
+    microvm.basic_config()
+
+    # Configure metrics based on a file.
+    metrics_path = os.path.join(microvm.path, 'metrics_fifo')
+    utils.run_cmd("touch {}".format(metrics_path))
+    response = microvm.metrics.put(
+        metrics_path=microvm.create_jailed_resource(metrics_path)
+    )
+    assert microvm.api_session.is_status_no_content(response.status_code)
+
+    microvm.start()
+
+    metrics_jail_path = os.path.join(microvm.jailer.chroot_path(),
+                                     metrics_path)
+    metrics_fd = open(metrics_jail_path)
+    line_metrics = metrics_fd.readlines()
+    print(line_metrics)
+    assert len(line_metrics) == 1
+
+    firecracker_pid = int(microvm.jailer_clone_pid)
+    size = os.path.getsize(metrics_jail_path)
+    res.prlimit(firecracker_pid, res.RLIMIT_FSIZE, (size*3, res.RLIM_INFINITY))
```

Comment:
Why `size *3` ?

Reply:
Because the metrics file is flushed twice and we have to make sure that the limit is bigger than that in order to make sure the SIGXFSZ metric is logged.